src/midi/midi_fluidsynth.cpp

```diff
@@ -128,19 +128,35 @@ static std::vector<std::string> get_data_dirs()
 
 static std::vector<std::string> get_data_dirs()
 {
+	// First priority is XDG_DATA_HOME
 	const char *xdg_data_home_env = getenv("XDG_DATA_HOME");
 	const auto xdg_data_home = CROSS_ResolveHome(
 	        xdg_data_home_env ? xdg_data_home_env : "~/.local/share");
 
-	return {
-	        CROSS_GetPlatformConfigDir() + "soundfonts/",
-	        xdg_data_home + "/soundfonts/",
-	        xdg_data_home + "/sounds/sf2/",
-	        "/usr/local/share/soundfonts/",
-	        "/usr/local/share/sounds/sf2/",
-	        "/usr/share/soundfonts/",
-	        "/usr/share/sounds/sf2/",
-	};
+	std::vector<std::string> dirs = {xdg_data_home + "/dosbox/soundfonts/",
```

Comment:
The container is no longer being filled once during initialization… Which means that this way of std::vector usage is not appropriate any more. Two ways to move through this:

1. replace std::vector with std::deque, or:
2. reserve enough space in vector (it is possible to pre-calculate how many strings will end up in it in this case)

Reply:
Went with deque for the directory lists.  Felt that keeping the code tighter (and avoiding pre-tally  code) is better for maintainers given there's zero performance concern here.